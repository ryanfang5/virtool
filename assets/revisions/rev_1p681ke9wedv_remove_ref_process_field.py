"""
Remove ref process field

Revision ID: 1p681ke9wedv
Date: 2022-06-09 22:17:02.297460

"""
import arrow
from motor.motor_asyncio import AsyncIOMotorClientSession, AsyncIOMotorDatabase

# Revision identifiers.
name = "Remove ref process field"
created_at = arrow.get("2022-06-09 22:17:02.297460")
revision_id = "1p681ke9wedv"
required_alembic_revision = None


async def upgrade(motor_db: AsyncIOMotorDatabase, session: AsyncIOMotorClientSession):
    await motor_db.references.update_many({}, {"$unset": {"process": ""}})


async def test_upgrade(mongo, snapshot):
    await mongo.references.insert_many(
        [{"_id": "foo", "process": "test"}, {"_id": "bar", "task": "test"}]
    )

    async with await mongo.client.start_session() as session:
        async with session.start_transaction():
            await upgrade(mongo, session)

    assert await mongo.references.find().to_list(None) == snapshot
