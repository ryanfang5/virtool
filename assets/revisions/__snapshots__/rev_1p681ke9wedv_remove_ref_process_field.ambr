# name: test_upgrade
  <class 'list'> [
    <class 'dict'> {
      '_id': 'foo',
    },
    <class 'dict'> {
      '_id': 'bar',
      'task': 'test',
    },
  ]
---
